src/overlays/actors/ovl_En_Po_Composer/z_en_po_composer.c

```diff
@@ -100,88 +128,618 @@ static DamageTable D_80BC67E8 = {
+                    }
+                    break;
+
+                default:
+                    break;
+            }
+        }
+
+        if (this->actionFunc != EnPoComposer_Appear) {
+            Cutscene_ActorTranslateAndYaw(&this->actor, play, cueChannel);
+        } else {
+            Cutscene_ActorTranslate(&this->actor, play, cueChannel);
+        }
+
+        EnPoComposer_StepLightAlpha(this);
+
+        if ((this->sharpCsNum == SHARP_CS_SONG_STORMS) && (play->csCtx.curFrame == 204)) {
+            Audio_PlaySfx(NA_SE_SY_DIZZY_EFFECT);
+        }
+
+        if (this->visible == true) {
+            Actor_PlaySfx_Flagged(&this->actor, NA_SE_EN_SHARP_FLOAT - SFX_FLAG);
+        }
+        return true;
+    }
+
+    if (this->actionFunc != EnPoComposer_StartedCutscene && this->actionFunc != EnPoComposer_StartCutscene &&
+        this->actionFunc != EnPoComposer_PlayCurse && play->csCtx.state == CS_STATE_IDLE) {
+
+        if (PO_COMPOSER_IS_FLAT(&this->actor)) {
+            EnPoComposer_SetupStartedCutscene(this);
+        } else if (PO_COMPOSER_4000(&this->actor)) {
+            EnPoComposer_SetupStartedCutscene(this);
+        } else {
+            this->inCutscene = false;
+
+            if (this->sharpCsNum < SHARP_CS_SONG_STORMS) {
+                if (!(CHECK_WEEKEVENTREG(WEEKEVENTREG_14_02))) {
+                    SET_WEEKEVENTREG(WEEKEVENTREG_14_02);
+                }
+                EnPoComposer_SetupPlayCurse(this);
+            } else {
+                EnPoComposer_SetupStartedCutscene(this);
+            }
+        }
+    }
+
+    this->lastCsAction = COMPOSER_CUEID_NONE;
+    return false;
+}
+
+void EnPoComposer_StepLightAlpha(EnPoComposer* this) {
+    if ((this->visible == true) && (this->lightColor.a != 255)) {
+        if (this->lightColor.a > 247) {
+            this->lightColor.a = 255;
+        } else {
+            this->lightColor.a += 7;
+        }
+    } else if (!this->visible && (this->lightColor.a != 0)) {
+        if (this->lightColor.a < 8) {
+            this->lightColor.a = 0;
+        } else {
+            this->lightColor.a -= 7;
+        }
+    }
+}
+
+void EnPoComposer_StepLightColor(EnPoComposer* this) {
+    s16 color;
+
+    color = this->lightColor.r + 5;
+    this->lightColor.r = CLAMP_MAX(color, 255);
+
+    color = this->lightColor.g + 5;
+    this->lightColor.g = CLAMP_MAX(color, 255);
+
+    if (this->lightColor.b > 210) {
+        color = this->lightColor.b - 5;
+        this->lightColor.b = CLAMP_MIN(color, 210);
+    } else {
+        color = this->lightColor.b + 5;
+        this->lightColor.b = CLAMP_MAX(color, 210);
+    }
+}
+
+void EnPoComposer_UpdateEnvColor(EnPoComposer* this) {
+    if ((this->actionFunc == EnPoComposer_Appear) && (this->skelAnime.curFrame < 12.0f)) {
+        this->envColor.r = this->envColor.g = this->envColor.b = 55.0f + this->skelAnime.curFrame * 16.66f;
+        this->envColor.a = this->skelAnime.curFrame * 16.666666f;
```

Comment:
I guess this doesn't match, right?
```suggestion
        this->envColor.a = this->skelAnime.curFrame * (50.0f / 3.0f);
```